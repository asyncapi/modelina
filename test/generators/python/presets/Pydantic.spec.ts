import {
  PythonGenerator,
  PYTHON_PYDANTIC_PRESET
} from '../../../../src/generators/python';

describe('PYTHON_PYDANTIC_PRESET', () => {
  let generator: PythonGenerator;

  beforeEach(() => {
    generator = new PythonGenerator({
      presets: [PYTHON_PYDANTIC_PRESET]
    });
  });

  test('should render pydantic for class', async () => {
    const doc = {
      title: 'Test',
      type: 'object',
      properties: {
        prop: {
          description: `test
  multi
  line
  description`,
          type: 'string'
        }
      }
    };

    const models = await generator.generate(doc);
    expect(models).toHaveLength(1);
    expect(models[0].result).toMatchSnapshot();
  });

  test('should render union to support Python < 3.10', async () => {
    const doc = {
      title: 'UnionTest',
      type: 'object',
      properties: {
        unionTest: {
          oneOf: [
            {
              title: 'Union1',
              type: 'object',
              properties: {
                testProp1: {
                  type: 'string'
                }
              }
            },
            {
              title: 'Union2',
              type: 'object',
              properties: {
                testProp2: {
                  type: 'string'
                }
              }
            }
          ]
        }
      }
    };

    const models = await generator.generate(doc);
    expect(models.map((model) => model.result)).toMatchSnapshot();
  });
});
