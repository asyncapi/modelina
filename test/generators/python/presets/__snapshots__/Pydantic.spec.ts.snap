// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PYTHON_PYDANTIC_PRESET should render pydantic for class 1`] = `
"class Test(BaseModel): 
  prop: Optional[str] = Field(alias='''test
    multi
    line
    description''', default=None)
  additionalProperties: Optional[dict[Any, Any]] = Field(default=None)
"
`;

exports[`PYTHON_PYDANTIC_PRESET should render union to support Python < 3.10 1`] = `
Array [
  "class UnionTest(BaseModel): 
  unionTest: Optional[Union[Union1, Union2]] = Field(default=None)
  additionalProperties: Optional[dict[Any, Any]] = Field(default=None)
",
  "class Union1(BaseModel): 
  testProp1: Optional[str] = Field(default=None)
  additionalProperties: Optional[dict[Any, Any]] = Field(default=None)
",
  "class Union2(BaseModel): 
  testProp2: Optional[str] = Field(default=None)
  additionalProperties: Optional[dict[Any, Any]] = Field(default=None)
",
]
`;
